test/test_time.cpp

```diff
@@ -26,6 +28,20 @@ inline void res( CLOCKID clk_id )
             << std::endl;
 }
 
+static void* test_thread_time(void *arg)
+{
+    int delay = *((int *) arg);
+
+    int c = 0;
+    while ( ++c < 10 )
+    {
+      usleep(delay);
+      std::cout << "[delay: " << delay << "] CLOCK_THREAD_CPUTIME_ID ("<< CLOCK_THREAD_CPUTIME_ID << ") " << time(CLOCK_THREAD_CPUTIME_ID) << std::endl;
```

Comment:
This clock measures how much CPU time a thread has used, so having two threads do exactly the same thing just with different length delays in between isn't going to produce a relevant difference.

Reply:
Both are sleeping, so it's not much of a difference. It would be much more informative to have one of the threads sleep every iteration, and the other do something CPU intensive for a while instead.

Reply:
Wait, are you thinking of the "shouldn't increment while sleeping" requirement of `CLOCK_UPTIME_RAW`? That's not relevant here since that refers to the entire system sleeping. But certainly a thread that does nothing but call `usleep` should have minimal CPU usage.

Reply:
I also had a similar issue with interpreting the results, as I am not sure they show (easily) what I would want to see.

What I would like to see is a much more direct test to confirm the CPU for a thread does not increment (significantly) whilst the tread is asleep. So for instance get the  CLOCK_THREAD_CPUTIME_ID before *and* after putting the thread to sleep, and report the difference between these. Then, run a number of threads with (very) different sleep times and see what we get. I would expect the increments in each to be basically the same, and not depend on the sleep time.

Reply:
OK, yeah, there are at least two things the tests could try to show:
 1. That a thread that sleeps for a significant time on each iteration doesn't measure significantly more CPU use than a thread that doesn't sleep at all (just prints the output), and
 2. That a thread that does lots of computation measures lots more CPU use than the above two cases.

The difficult thing is quantifying just how close or far apart the numbers should be.